keys.go

```diff
@@ -260,6 +272,41 @@ func (c *Client) SetKeyRing(ctx context.Context, idOrAlias, newKeyRingID string)
 	return &response.Keys[0], nil
 }
 
+// CreateKeyWithPolicyOverride creates a new KP key with given key policy details
+func (c *Client) CreateKeyWithPolicyOverride(ctx context.Context, name string, expiration *time.Time, extractable bool, aliases []string, policy Policy) (*Key, error) {
+	return c.CreateImportedKeyWithPolicyOverride(ctx, name, expiration, "", "", "", extractable, aliases, policy)
+}
+
+// CreateImportedKeyWithPolicyOverride creates a new Imported KP key from the given key material and with given key policy details
+func (c *Client) CreateImportedKeyWithPolicyOverride(ctx context.Context, name string, expiration *time.Time, payload, encryptedNonce, iv string, extractable bool, aliases []string, policy Policy) (*Key, error) {
+	key := c.createKeyTemplate(ctx, name, expiration, payload, encryptedNonce, iv, extractable, aliases, AlgorithmRSAOAEP256)
+
+	key.Rotation = policy.Rotation
```

Comment:
createKeyTemplate() function is meant to form the payload. So, it makes sense to change the signature of this function to accept policy also and populate the policy along with other attributes like aliases, import key attributes etc., Since it is a private function, changing the signaure doesn't affect anything.. the createKey() flows can pass nil to policies whereas overrides flow can send the policy data